react/State.js

```diff
@@ -0,0 +1,192 @@
+        })
+        .map(mapToMinicartItem)
+
+      cache.writeData({
+        data: {
+          minicart: { __typename: 'Minicart', items },
+        },
+      })
+      return items
+    },
+    updateOrderForm: (_, { orderForm: newOrderForm }, { cache }) => {
+      const orderForm = mapToLinkStateOrderForm(newOrderForm)
+      cache.writeData({
+        data: {
+          minicart: { __typename: 'Minicart', orderForm },
+        },
+      })
+      if (orderForm.items) {
+        updateLinkStateItems(orderForm.items, cache)
+      }
+      return orderForm
+    },
+  },
+}
+
+function updateLinkStateItems(newItems, cache) {
+  const items = newItems.map(item =>
+    mapToMinicartItem({
+      upToDate: true,
+      ...item,
+    })
+  )
+
+  cache.writeData({
+    data: {
+      minicart: { __typename: 'Minicart', items },
+    },
+  })
+  return items
+}
+
+const mapToLinkStateOrderForm = orderForm => ({
+  ...orderForm,
+  __typename: 'OrderFormClient',
+  items: orderForm.items && orderForm.items.map(mapToOrderFormItem),
+  totalizers:
+    orderForm.totalizers && orderForm.totalizers.map(mapToOrderFormTotalizer),
+  clientProfileData: orderForm.clientProfileData && {
+    ...orderForm.clientProfileData,
+    __typename: 'OrderFormClientProfileData',
+  },
+  shippingData:
+    orderForm.shippingData &&
+    mapToOrderFormShippingData(orderForm.shippingData),
+  storePreferencesData:
+    orderForm.storePreferencesData &&
+    mapToOrderFormStorePreferences(orderForm.storePreferencesData),
+})
+
+const mapAssemblyOptions = options =>
+  options && {
+    ...options,
+    added:
+      options.added &&
+      options.added.map(added => ({
+        item: added.item && mapToOrderFormItem(added.item),
+        ...added,
+        __typename: 'AddedAssemblyOptions',
+      })),
+    removed:
+      options.removed &&
+      options.removed.map(option => ({
+        ...option,
+        __typename: 'RemovedAssemblyOption',
+      })),
+    __typename: 'AssemblyOptions',
+  }
+
+const mapToOrderFormItem = item => ({
+  ...item,
+  assemblyOptions: mapAssemblyOptions(item.assemblyOptions),
+  __typename: 'OrderFormItem',
+})
+
+const mapToOrderFormTotalizer = totalizer => ({
+  ...totalizer,
+  __typename: 'OrderFormTotalizer',
+})
+
+const mapToOrderFormShippingData = ({ address, availableAddresses }) => ({
+  address: address && mapToAddress(address),
+  availableAddresses:
+    availableAddresses && availableAddresses.map(mapToAddress),
+  __typename: 'OrderFormShippingData',
+})
+
+const mapToAddress = address => ({
+  ...address,
+  __typename: 'Address',
+})
+
+const mapToOrderFormStorePreferences = storePreferencesData => ({
+  ...storePreferencesData,
+  __typename: 'OrderFormStorePreferencesData',
+})
+
+const mapToMinicartItem = item => ({
+  seller: null,
+  index: null,
+  parentItemIndex: null,
+  parentAssemblyBinding: null,
+  ...item,
+  __typename: 'MinicartItem',
+})
+
+export const initialState = {
```

Comment:
I would prefer you to assign initialState and resolvers to variables, and exporting them at the end of the file